Compile/interface/ZJetProduct.h

```diff
@@ -181,10 +186,22 @@ class ZJetProduct : public KappaProduct
                      unsigned long index,
                      std::string corrLevel) const
     {
-        assert(GetInvalidJetCount(settings, event, corrLevel) > index);
+        if (GetInvalidJetCount(settings, event, corrLevel) <= index) {
+            return 0;
+        }
 
-        // Invalid jets, no need for different correction levels
-        return static_cast<KLV*>(m_invalidJets[index]);
+        // Gen jets are always valid
+        if (corrLevel == "Gen") {
+            return 0;
+        }
+        // Uncorrected invalid jet
+        else if (corrLevel == "None") {
+            return static_cast<KLV*>(m_invalidJets[index]);
```

Comment:
is this really correct?
eventually this is implicitely casted to usigned long when returning?